app/controllers/quick_actions_controller.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class QuickActionsController < ApplicationController
+  before_action :authenticate_user!
+  load_resource :facility, find_by: :url_name
+  load_resource :instrument, through: :facility, find_by: :url_name
+  before_action :get_startable_reservation, only: [:index, :update]
+  before_action :prepare_reservation_data, only: [:new, :create]
+
+  # GET /facilities/:facility_id/instruments/:instrument_id/quick_actions
+  def index
+    redirect_to(new_facility_instrument_quick_action_path) unless @startable || @ongoing
+  end
+
+  def new
+    @walkup_available = @instrument.walkup_available?(interval: @reservation_intervals.first)
+  end
+
+  def create
+    index = reservation_params[:reservation_index].to_i
+    account = Account.find(reservation_params[:account])
+    reservation = @possible_reservations[index]
+
+    build_order
+
+    # Borrowed from ReservationCreator
+    @order.account = account
+    validator = OrderPurchaseValidator.new(@order_detail)
+    raise OrderPurchaseValidatorError, @order_detail if validator.invalid?
+    reservation.save_as_user!(current_user)
+    @order_detail.assign_estimated_price(reservation.reserve_end_at)
+    @order_detail.save_as_user!(current_user)
+
+    reservation.start_reservation!
+    redirect_to facility_instrument_quick_actions_path(@facility, @instrument)
+  end
+
+  def update
```

Comment:
Is this a WIP?  If not, `start` might be a more intention-revealing name for the action. I'd expect `update` to be handling params